OData/src/System.Web.OData/OData/Formatter/Serialization/ODataEntityTypeSerializer.cs

```diff
@@ -465,10 +465,20 @@ public virtual ODataNavigationLink CreateNavigationLink(IEdmNavigationProperty n
             List<ODataProperty> properties = new List<ODataProperty>();
             foreach (IEdmStructuralProperty structuralProperty in structuralProperties)
             {
-                ODataProperty property = CreateStructuralProperty(structuralProperty, entityInstanceContext);
-                if (property != null)
+                ODataSerializerContext writeContext = entityInstanceContext.SerializerContext;
+                object propertyValue = entityInstanceContext.GetPropertyValue(structuralProperty.Name);
+
+                var config = writeContext.Request.GetConfiguration();
+                bool doNotSerializeIfNull = config.GetDoNotSerializeNullCollections();
+
+                bool skipEntireCollectionSerialization = (propertyValue == null && structuralProperty.Type.Definition.TypeKind == EdmTypeKind.Collection && doNotSerializeIfNull);
+                if (!skipEntireCollectionSerialization)
```

Comment:
Look at where the variable is consumed, in the IF statement, I can't implement "DoNotSerializeNullCollections" if this code is removed.  If I remove this code, it will serialize null collections as empty.  I do already have code in the CreateODataValue method to handle "SerializeNullCollectionsAsEmpty" but that is a separate option.  Removing this code would break "DoNotSerializeNullCollections".  If you want this code removed, you are implicitly asking me to *remove* the option "DoNotSerializeNullCollections".  Is this your ask?